measurement/tidb/pdctl.py

```diff
@@ -74,11 +74,17 @@ def build_url(uri):
     def save_info(self, basedir=None):
         full_outputdir = fileutils.build_full_output_dir(
             basedir=basedir, subdir=self.pdctl_dir)
-        fileutils.write_file(os.path.join(
-            full_outputdir, "%s-health.json" % self.pd_host), self.read_health())
-        fileutils.write_file(os.path.join(
-            full_outputdir, "%s-diagnose.json" % self.pd_host), self.read_diagnose())
+        pd_health = self.read_health()
+        if pd_health:
+            fileutils.write_file(os.path.join(
+                full_outputdir, "%s-health.json" % self.pd_host), pd_health)
+        pd_diagnose = self.read_diagnose()
+        if pd_diagnose:
+            fileutils.write_file(os.path.join(
+                full_outputdir, "%s-diagnose.json" % self.pd_host), self.read_diagnose())
```

Comment:
Is self.read_diagnose() => pd_diagnose better ?

Reply:
It's under class `PDCtl` so I think it's not necessary to label `pd_` again in method names.

Reply:
Sorry, I didn't express it clearly.  Well I mean `self.read_diagnose()` has been called at L81, so I think it's not necessary to call this method twice at L84.

Reply:
Ah, that was missed in the commit, just pushed a fix.